python/cudf_polars/cudf_polars/experimental/distinct.py

```diff
@@ -100,20 +104,11 @@ def lower_distinct(
             # partitions. For now, we raise an error to fall back
             # to one partition.
             raise NotImplementedError("Unsupported slice for multiple partitions.")
-    else:
+    elif cardinality is not None:
         # Use cardinality to determine partitioning
-        assert config_options.executor.name == "streaming", (
-            "'in-memory' executor not supported in 'lower_distinct'"
-        )
-        cardinality_factor = {
-            c: min(f, 1.0)
-            for c, f in config_options.executor.cardinality_factor.items()
-            if c in subset
-        }
-        if cardinality_factor:
-            cardinality = max(cardinality_factor.values())
-            n_ary = min(max(int(1.0 / cardinality), 2), child_count)
-            output_count = max(int(cardinality * child_count), 1)
+        cardinality = max(cardinality, 0.0001)  # Avoid divide by 0
```

Comment:
This is a good point, not sure, but would it make it more sense to just replace `0.0` with `0.0001` in both https://github.com/rapidsai/cudf/pull/18725/files#diff-91044f2adf9fac866e8511f93376be08a1954f572c9b314cf3b0910022a98082R161 and https://github.com/rapidsai/cudf/pull/18725/files#diff-c93704a29e1c3263ea7267bf2028f187f43d40727bda1b8ffbcdd87076bac77bR179 instead?